core/nandtool/flashmng.c

```diff
@@ -81,6 +108,39 @@ int flashmng_erase(oid_t oid, unsigned int start, unsigned int size)
 }
 
 
+/* write JFFS2 clean block markers */
+int flashmng_cleanMarkers(oid_t oid, unsigned int start, unsigned int size)
+{
+	void *metabuf;
+	int ret = 0;
+	unsigned int blockno;
+	flashsrv_info_t *info = flashmng_info(oid);
+
+	if (info == NULL)
+		return -EFAULT;
+
+	metabuf = malloc(info->writesz);
+
+	if (!metabuf)
+		return -ENOMEM;
+
+	memset(metabuf, 0xff, info->writesz);
+	memcpy(metabuf, &oob_cleanmarker, sizeof(oob_cleanmarker));
+
+	for (blockno = start; blockno < start + size; ++blockno) {
+		uint32_t addr = blockno * info->erasesz;
+		if (flashmng_isbad(oid, blockno)) {
+			printf("cleanMarkers: block %u is marked as bad - skipping\n", blockno);
+			continue;
+		}
+		ret += writemeta(oid, addr, metabuf, info->writesz);
```

Comment:
```suggestion
		if (flashmng_isbad(oid, blockno)) {
			printf("cleanMarkers: block %u is marked as bad - skipping\n", blockno);
			continue;
		}
		ret += writemeta(oid, blockno * info->erasesz, metabuf, info->writesz);
```

Reply:
preferred to highlight that `writemeta()` takes page **address** as opposed to `writeraw()` which takes page **number**, but changed assuming anyone digging around this code would need to know `page_addr = eraseblock_no * erasesize`